src/lib.rs

```diff
@@ -86,3 +89,43 @@ impl TransmutingByteSlices for [u8] {
         unsafe { self.align_to::<u32>() }
     }
 }
+
+// Provide `[u32; 4]` and `[u64; 4]` iteration support for unaligned buffer.
```

Comment:
It's a bit over-cautious, but I think I'd like to see a marker trait for the two specific types that we care about and have thought through, as opposed to just any `T`. This should be an `unsafe` trait and the implementation can be a place we can keep the unsafe rationale.

Reply:
That's a good idea.